### PR TITLE
feat(esp32_usb_stream): update usb stream lib

### LINKED ISSUE #2: Both UVC & UAC getting start doesn't want to compile for ESP32-S3
```
Compiling core...
/usr/bin/env bash -c "echo -DARDUINO_CORE_BUILD > '/var/folders/kg/wxdt75kd0lvcb91rmqvck1f40000gn/T/arduino_build_15691/file_opts'"
Using precompiled core: /var/folders/kg/wxdt75kd0lvcb91rmqvck1f40000gn/T/arduino_cache_245035/core/core_4b6716b556e99d089fdbecee560f77aa.a
/usr/bin/env bash -c ": > '/var/folders/kg/wxdt75kd0lvcb91rmqvck1f40000gn/T/arduino_build_15691/file_opts'"
Linking everything together...
/Users/joel/Library/Arduino15/packages/esp32/tools/esp-xs3/2302/bin/xtensa-esp32s3-elf-g++ @/Users/joel/Library/Arduino15/packages/esp32/tools/esp32-arduino-libs/idf-release_v5.1-442a798083/esp32s3/flags/ld_flags @/Users/joel/Library/Arduino15/packages/esp32/tools/esp32-arduino-libs/idf-release_v5.1-442a798083/esp32s3/flags/ld_scripts -Wl,--Map=/var/folders/kg/wxdt75kd0lvcb91rmqvck1f40000gn/T/arduino_build_15691/GettingStartUVC.ino.map -L/Users/joel/Library/Arduino15/packages/esp32/tools/esp32-arduino-libs/idf-release_v5.1-442a798083/esp32s3/lib -L/Users/joel/Library/Arduino15/packages/esp32/tools/esp32-arduino-libs/idf-release_v5.1-442a798083/esp32s3/ld -L/Users/joel/Library/Arduino15/packages/esp32/tools/esp32-arduino-libs/idf-release_v5.1-442a798083/esp32s3/qio_qspi -Wl,--wrap=esp_panic_handler -Wl,--start-group /var/folders/kg/wxdt75kd0lvcb91rmqvck1f40000gn/T/arduino_build_15691/sketch/GettingStartUVC.ino.cpp.o /var/folders/kg/wxdt75kd0lvcb91rmqvck1f40000gn/T/arduino_build_15691/libraries/ESP32_USB_STREAM/USB_STREAM.cpp.o /var/folders/kg/wxdt75kd0lvcb91rmqvck1f40000gn/T/arduino_build_15691/libraries/ESP32_USB_STREAM/original/descriptor.c.o /var/folders/kg/wxdt75kd0lvcb91rmqvck1f40000gn/T/arduino_build_15691/libraries/ESP32_USB_STREAM/original/usb_host_helpers.c.o /var/folders/kg/wxdt75kd0lvcb91rmqvck1f40000gn/T/arduino_build_15691/libraries/ESP32_USB_STREAM/original/usb_stream.c.o /var/folders/kg/wxdt75kd0lvcb91rmqvck1f40000gn/T/arduino_cache_245035/core/core_4b6716b556e99d089fdbecee560f77aa.a @/Users/joel/Library/Arduino15/packages/esp32/tools/esp32-arduino-libs/idf-release_v5.1-442a798083/esp32s3/flags/ld_libs -Wl,--end-group -Wl,-EL -o /var/folders/kg/wxdt75kd0lvcb91rmqvck1f40000gn/T/arduino_build_15691/GettingStartUVC.ino.elf
/Users/joel/Library/Arduino15/packages/esp32/tools/esp-xs3/2302/bin/../lib/gcc/xtensa-esp32s3-elf/12.2.0/../../../../xtensa-esp32s3-elf/bin/ld: /var/folders/kg/wxdt75kd0lvcb91rmqvck1f40000gn/T/arduino_build_15691/libraries/ESP32_USB_STREAM/original/usb_stream.c.o:(.literal.usb_streaming_start+0x34): undefined reference to `mps_limits_default'
/Users/joel/Library/Arduino15/packages/esp32/tools/esp-xs3/2302/bin/../lib/gcc/xtensa-esp32s3-elf/12.2.0/../../../../xtensa-esp32s3-elf/bin/ld: /var/folders/kg/wxdt75kd0lvcb91rmqvck1f40000gn/T/arduino_build_15691/libraries/ESP32_USB_STREAM/original/usb_stream.c.o:(.literal.usb_streaming_start+0x6c): undefined reference to `mps_limits_bias_rx'
collect2: error: ld returned 1 exit status
```

Highlights on 2 errors:
undefined reference to 'mps_limits_bias_rx'
undefined reference to 'mps_limits_default'

### PR DESCRIPTION
**Pull request by lijunru-hub**
Closes [2!](https://github.com/esp-arduino-libs/ESP32_USB_Stream/issues/2)

1. Supports compilation with idf5.1.
2. Synchronize code to uvc_steam version 1.4.0.

### CODE REVIEW

**Review by leeebo — APPROVED** [review]
@lijunru-hub LGTM!

**Comment by lijunru-hub** [comment]
@Lzw655 PTAL

File: idf_component.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+version: "1.4.0"
```
Comment (Lzw655) [inline idf_component.yml]:
```suggestion
version: "0.1.0"
```

File: library.properties [hunk 2]
```diff
@@ -1,7 +1,7 @@
 name=ESP32_USB_STREAM
-version=0.0.1
+version=1.4.0
```
Comment (Lzw655) [inline library.properties]:
```suggestion
version=0.1.0
```

**Review by Lzw655 — APPROVED** [review]
Please also update the README. Thank you!

File: idf_component.yml (hunk 1 above)
Comment (lijunru-hub) [reply idf_component.yml]: 这个是 USB_Stream 的源文件的版本号，不是 arduino 的，这个就不改了

File: library.properties (hunk 2 above)
Comment (lijunru-hub) [reply library.properties]: 已改

**Pull request merged**